app/protective-monitoring/publish-event.js

```diff
@@ -0,0 +1,36 @@
+const messageSchema = require('./message-schema')
+const wreck = require('wreck')
+
+class PublishEvent {
+  constructor (endPoint, log = true) {
+    this.endPoint = endPoint
+    this.log = log
+  }
+
+  async sendEvent (jsonMessage) {
+    try {
+      if (this.endPoint === null || this.endPoint === '') {
```

Comment:
I think you could simplify the logic flow of the if/else and try/catch to something like this:

```
if (this.endPoint) {
      const validationResult = messageSchema.validate(jsonMessage)

      if (validationResult.error) {
        throw new Error(`Protective monitoring event schema is invalid. ${validationResult.error.message}`)
      }

      try {
        await wreck.post(this.endPoint, {
          payload: jsonMessage,
          json: true
        })
      } catch (error) {
        console.error(`Unable to send protective monitoring event. ${error.message}`)
      }
    } else if (this.log) {
      console.log('Protective monitoring endpoint not set')
    }
  }
```
Not having the endpoint set is going to be very common in local development and don't want to flood the logs with errors.

Also suggest making the logs more specific to protective monitoring as if teams are viewing these logs through App Insights, it might be difficult to filter/quickly interpret which JSON schema is invalid when diagnosing issues in live services.

Reply:
Agreed and changed.